### DISCUSSION TITLE
Crossorigin preloads despite assets being hosted on the same origin

### DISCUSSION THREAD

**Discussion by klapec** [vercel/next.js#13442] [Help] [answered]
Hi,

I noticed that since switching the `experimental.modern` option on (in `next.config.js`) all the `<link rel="preload"...`-s got the `crossorigin="anonymous"` attribute (https://github.com/zeit/next.js/blob/canary/packages/next/build/webpack-config.ts#L464). What's the reasoning behind this? I'm having my assets hosted on the same origin as the page itself, is this `crossorigin` attribute necessary?

Thanks.

**Comment by klapec** [comment]
Also, shouldn't preloads for `type="module"` scripts use `<link rel="modulepreload"` instead, as per https://developers.google.com/web/updates/2017/12/modulepreload?

**Comment by klapec** [comment]
Bump, anyone? It's a bummer that I had to disable `experimental.modern` because of this - my project's CDN seems to be configured incorrectly when it comes to responding with proper CORS headers and I don't have an access to it.

**Comment by janicklas-ralph** [comment, marked as answer]
Hi @klapec please see this comment explaining the reason behind setting `crossorigin=anonymous`

https://github.com/vercel/next.js/issues/9459#issuecomment-556244044

As to why we don't use `rel="modulepreload"` is because it's not yet supported in all browsers - https://caniuse.com/#search=modulepreload

**Reply by klapec** [reply]
Thank you. Seems a bit weird to me that same-origin requests require cross-origin headers to work, but oh well ;)

**Reply by janicklas-ralph** [reply]
In the near future, we will revisit `modulepreload` again as the support increases.

**Reply by klapec** [reply]
I saw you asked for the before/after results for `experimental.modern` on my build but you've edited your comment - I've got some screenshots nevertheless ;)

before:
<img width="1032" alt="before" src="https://user-images.githubusercontent.com/565022/84111785-ede6e500-aa27-11ea-887c-917e1e6e0233.png">

after:
<img width="1032" alt="after" src="https://user-images.githubusercontent.com/565022/84111825-f6d7b680-aa27-11ea-8f32-8ad2b44a53ba.png">

**Reply by janicklas-ralph** [reply]
Thank you! This helps.
I wasn't sure if it was appropriate to ask/share on public :)
